src/RestWrite.js

```diff
@@ -281,26 +281,26 @@ RestWrite.prototype.handleAuthData = function(authData) {
 
     this.storage['authProvider'] = Object.keys(authData).join(',');
 
-    if (results.length == 0) {
-      this.data.username = cryptoUtils.newToken();
```

Comment:
just needed to remove that so we don't override username when we have no results